src/Neo/SmartContract/Native/PolicyContract.cs

```diff
@@ -131,6 +153,21 @@ public uint GetStoragePrice(IReadOnlyStore snapshot)
         }
 
         /// <summary>
+        /// Gets the block generation time in milliseconds.
+        /// This can only be called after the HF_Echidna.
+        /// </summary>
+        /// <param name="snapshot">The snapshot used to read data.</param>
+        /// <returns>The block generation time in milliseconds.</returns>
+        [ContractMethod(Hardfork.HF_Echidna, CpuFee = 1 << 15, RequiredCallFlags = CallFlags.ReadStates)]
+        public uint GetBlockGenTime(IReadOnlyStore snapshot)
```

Comment:
I'd also propose to rename it to `MSPerBlock` or `MillisecondsPerBlock`, the reason is in https://github.com/nspcc-dev/neo-go/pull/3835#discussion_r1991409587.